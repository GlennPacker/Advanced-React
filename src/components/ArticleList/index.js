import React from 'react';
import Article from '../Article';

const ArticleList = ({ articles, articleActions }) => {
	return (
		<div>
			{(Object.values(articles)).map(article => {
				return (
					<Article
						article={article}
						actions={articleActions}
						key={article.id}
					/>
				)
			})}
		</div>
	);
}

export default ArticleList;
