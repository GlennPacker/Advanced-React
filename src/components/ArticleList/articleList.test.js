import React from 'react';
import ArticleList from './index';

import renderer from 'react-test-renderer';

describe('ArticleList Component', () => {

	const testProps = {
		articles: {
			a: { id: 1 },
			b: { id: 2 }
		},
		articleActions: {
			lookupAuthor: jest.fn(() => ({}))
		}
	}

	it('renders connrectly', () => {
		const ele = renderer.create(
			<ArticleList
				{...testProps}
			/>
		)
		expect(ele).toMatchSnapshot();
	})
})