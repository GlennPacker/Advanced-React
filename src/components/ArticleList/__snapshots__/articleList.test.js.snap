// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Component renders connrectly 1`] = `
<div>
  <div />
  <div />
  <div>
    <a>
       
    </a>
  </div>
  <div />
  <hr />
  <div />
  <div />
  <div>
    <a>
       
    </a>
  </div>
  <div />
  <hr />
</div>
`;
